<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsDialog</class>
 <widget class="QDialog" name="SettingsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>400</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>400</width>
    <height>300</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="tabShape">
      <enum>QTabWidget::Rounded</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="networkTab">
      <attribute name="title">
       <string>Network</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="serverSettingsGroupBox">
         <property name="title">
          <string>Server Settings</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="portLabel">
            <property name="text">
             <string>Port</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="interfaceLabel">
            <property name="text">
             <string>Interface</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QComboBox" name="interfaceComboBox"/>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="portSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="value">
             <number>8049</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="connectionsGroupBox">
         <property name="title">
          <string>Connections</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QTreeView" name="connectionsList">
            <property name="editTriggers">
             <set>QAbstractItemView::NoEditTriggers</set>
            </property>
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <property name="selectionMode">
             <enum>QAbstractItemView::ExtendedSelection</enum>
            </property>
            <property name="rootIsDecorated">
             <bool>false</bool>
            </property>
            <property name="itemsExpandable">
             <bool>false</bool>
            </property>
            <property name="allColumnsShowFocus">
             <bool>true</bool>
            </property>
            <property name="expandsOnDoubleClick">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="disconnectButton">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="toolTip">
               <string>Disconnect the selected clients</string>
              </property>
              <property name="text">
               <string>Disconnect</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="activityTab">
      <attribute name="title">
       <string>Activity</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QGroupBox" name="messagesGroupBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Notifications</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="2" column="0" colspan="4">
           <widget class="QCheckBox" name="disconnectCheckBox">
            <property name="text">
             <string>Notify me when a client disconnects</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="4">
           <widget class="QCheckBox" name="connectCheckBox">
            <property name="text">
             <string>Notify me when a client connects</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="durationLabel">
            <property name="text">
             <string>Duration:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="durationSpinBox">
            <property name="suffix">
             <string> s</string>
            </property>
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>1800</number>
            </property>
            <property name="value">
             <number>5</number>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QLabel" name="explainationLabel">
            <property name="text">
             <string>(How long to display notification)</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="activationGroupBox">
         <property name="title">
          <string>Connection to Music Player</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0" colspan="3">
           <widget class="QCheckBox" name="activationCheckBox">
            <property name="text">
             <string>Reconnect to player when commands are received</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="disconnectPlayerButton">
            <property name="toolTip">
             <string>Disconnect from iTunes</string>
            </property>
            <property name="text">
             <string>Disconnect</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="connectPlayerButton">
            <property name="toolTip">
             <string>Connect to iTunes</string>
            </property>
            <property name="text">
             <string>Connect</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="securityTab">
      <attribute name="title">
       <string>Security</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QGroupBox" name="pinGroupBox">
         <property name="title">
          <string>Require PIN Authentication</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="pinLabel">
            <property name="text">
             <string>PIN:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="pinLineEdit">
            <property name="text">
             <string>12345</string>
            </property>
            <property name="maxLength">
             <number>32</number>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="logTab">
      <attribute name="title">
       <string>Log</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QGroupBox" name="eventLogGroupBox">
         <property name="title">
          <string>Event Log</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="0" column="0" colspan="5">
           <widget class="QTextEdit" name="logTextEdit">
            <property name="undoRedoEnabled">
             <bool>false</bool>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="4">
           <widget class="QPushButton" name="saveLogButton">
            <property name="toolTip">
             <string>Save log to file</string>
            </property>
            <property name="text">
             <string>Save</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="maxLogEntriesLabel">
            <property name="text">
             <string>Max Log Entries:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="maxLogEntriesSpinBox">
            <property name="suffix">
             <string/>
            </property>
            <property name="minimum">
             <number>8</number>
            </property>
            <property name="maximum">
             <number>4096</number>
            </property>
            <property name="value">
             <number>64</number>
            </property>
           </widget>
          </item>
          <item row="2" column="2" colspan="2">
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="logAllCheckBox">
         <property name="text">
          <string>Log all communication with clients</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Apply|QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
  <action name="actionSettings">
   <property name="text">
    <string>Show &amp;Settings Dialog</string>
   </property>
   <property name="toolTip">
    <string>Show Settings Dialog</string>
   </property>
   <property name="font">
    <font>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
  </action>
  <action name="actionConnect">
   <property name="text">
    <string>&amp;Connect to iTunes</string>
   </property>
  </action>
  <action name="actionDisconnect">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Disconnect from iTunes</string>
   </property>
   <property name="toolTip">
    <string>Disconnect from iTunes</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>portSpinBox</tabstop>
  <tabstop>interfaceComboBox</tabstop>
  <tabstop>connectionsList</tabstop>
  <tabstop>disconnectButton</tabstop>
  <tabstop>buttonBox</tabstop>
  <tabstop>tabWidget</tabstop>
  <tabstop>connectCheckBox</tabstop>
  <tabstop>disconnectCheckBox</tabstop>
  <tabstop>durationSpinBox</tabstop>
  <tabstop>activationCheckBox</tabstop>
  <tabstop>connectPlayerButton</tabstop>
  <tabstop>disconnectPlayerButton</tabstop>
  <tabstop>pinGroupBox</tabstop>
  <tabstop>pinLineEdit</tabstop>
  <tabstop>logTextEdit</tabstop>
  <tabstop>maxLogEntriesSpinBox</tabstop>
  <tabstop>saveLogButton</tabstop>
  <tabstop>logAllCheckBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>portSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>SettingsDialog</receiver>
   <slot>serverSettingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>116</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>83</x>
     <y>65</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>interfaceComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>SettingsDialog</receiver>
   <slot>serverSettingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>367</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>288</x>
     <y>65</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>disconnectButton</sender>
   <signal>clicked()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>disconnectButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>366</x>
     <y>311</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>clicked(QAbstractButton*)</signal>
   <receiver>SettingsDialog</receiver>
   <slot>apply(QAbstractButton*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>390</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>282</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>367</x>
     <y>390</y>
    </hint>
    <hint type="destinationlabel">
     <x>397</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>277</x>
     <y>390</y>
    </hint>
    <hint type="destinationlabel">
     <x>103</x>
     <y>276</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSettings</sender>
   <signal>triggered()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>showWindow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionConnect</sender>
   <signal>triggered()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>connectController()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDisconnect</sender>
   <signal>triggered()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>disconnectController()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>connectPlayerButton</sender>
   <signal>clicked()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>connectController()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>223</x>
     <y>206</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>234</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>disconnectPlayerButton</sender>
   <signal>clicked()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>disconnectController()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>342</x>
     <y>209</y>
    </hint>
    <hint type="destinationlabel">
     <x>392</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maxLogEntriesSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>SettingsDialog</receiver>
   <slot>setMaxLogEntries(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>129</x>
     <y>303</y>
    </hint>
    <hint type="destinationlabel">
     <x>7</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>serverSettingsChanged()</slot>
  <slot>disconnectButtonClicked()</slot>
  <slot>apply(QAbstractButton*)</slot>
  <slot>connectController()</slot>
  <slot>disconnectController()</slot>
  <slot>setMaxLogEntries(int)</slot>
 </slots>
</ui>
